tests/php/json-api/test-class.json-api-update-post-endpoints.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * Jetpack `sites/%s/posts/%d` endpoint unit tests.
+ *
+ * @package Jetpack
+ */
+
+require_jetpack_file( 'class.json-api-endpoints.php' );
+
+/**
+ * Jetpack `sites/%s/posts/%d` endpoint unit tests.
+ */
+class WP_Test_Json_Api_Update_Post_Endpoints extends WP_UnitTestCase {
+
+	/**
+	 * Inserts globals needed to initialize the endpoint.
+	 */
+	private function set_globals() {
+		$_SERVER['REQUEST_METHOD'] = 'Post';
+		$_SERVER['HTTP_HOST']      = '127.0.0.1';
+		$_SERVER['REQUEST_URI']    = '/';
+	}
+
+	/**
+	 * Prepare the environment for the test.
+	 */
+	public function setUp() {
+		global $blog_id;
+
+		if ( ! defined( 'WPCOM_JSON_API__BASE' ) ) {
+			define( 'WPCOM_JSON_API__BASE', 'public-api.wordpress.com/rest/v1' );
+		}
+
+		parent::setUp();
+
+		$this->set_globals();
+
+		WPCOM_JSON_API::init()->token_details = array( 'blog_id' => $blog_id );
+	}
+
+	/**
+	 * Unit tests the should untrash post protected method.
+	 */
+	public function test_update_post_api_v1_1() {
```

Comment:
In this specific test, we are testing `should_untrash_post ` method. I feel that the test name is a bit vague.

Reply:
Just update the test method name. Thanks for pointing that out.